grails-app/services/org/pih/warehouse/invoice/PrepaymentInvoiceService.groovy

```diff
@@ -0,0 +1,69 @@
+package org.pih.warehouse.invoice
+
+import org.pih.warehouse.core.Constants
+import org.pih.warehouse.order.Order
+import org.pih.warehouse.order.OrderAdjustment
+import org.pih.warehouse.order.OrderItem
+import org.pih.warehouse.order.OrderItemStatusCode
+import org.pih.warehouse.shipping.ShipmentItem
+import grails.gorm.transactions.Transactional
+
+@Transactional
+class PrepaymentInvoiceService {
+
+    InvoiceService invoiceService
```

Comment:
Generally my preference would be to not have services call other services but instead move common code to shared util classes (Ex: InvoiceNumberGenerator, InvoiceItemGenerator (or whatever names)). It breaks the code up into more manageable units (easier testing too).

I don't expect you to to make any kind of change like that here though. Just noting down my thoughts.